docs/architecture/pairing-adapter-for-integration-system.md

```diff
@@ -0,0 +1,29 @@
+
+## Solution overview
+
+This diagram illustrates the flow for pairing the Application with its External Broker using the Pairing Adapter.
+
+![External Broker Pairing Adapter](./assets/external-broker-pairing-adapter.svg)
+
+>**NOTE** The Application must be created before the pairing is performed. To create an Application that is managed by the Integration System the Application should be created using Application Template.
+
+### Steps
+
+1. The Administrator requests the pairing data for the Application using the Cockpit (1).
+
+1. The Cockpit contacts the Director component to obtain the pairing data for the Application (2). The Application is loosely coupled with the Integration System that manages it. Thanks to this relation the Director component is aware which Pairing Adapter is assigned to the Integration System. Knowing this, the Pairing Adapter is called to request the pairing data (3).
+
+1. The Pairing Adapter calls External Broker to obtain pairing data (4).
+
+1. Administrator after obtaining the pairing data provides them to the Application instance which allows the Application instance to establish trusted relation with External Broker (6). This informs External Broker that there is a new Application instance that needs to be managed via this External Broker.
+
+## Details
+
+The Compass supports multiple different External Brokers (represented internally by Integration Systems) to help abstract the way how Applications and Runtimes can be managed. Each Integration System can have its own approach for obtaining details for establishing a trusted relation for the managed system. To simplify the process and improve users' experience we introduce the concept of Pairing Adapters. Each Integration System can be coupled with Pairing Adapter. Thanks to this approach the Directors' API can leverage different Integration Systems under single mutation and handle the requests in a consistent manner. When Administrator wants to establish a new trusted relation between Application or Runtime and Integration System the API offers `generateAuthForApplication(ID, type)` and `generateAuthForRuntime(ID, type)` mutations.
+
+Both those mutations abstract the way of obtaining trusted relation data from the Integration System. Calling one of those mutations with appropriate parameters checks whether the desired system (Runtime or Application) is under control of the Integrations System. If the underlying Integration System has Pairing Adapter assigned, the call is enriched with additional data defined for Integration System and data which comes from the request context. Next, the request is handed to the Pairing Adapter. The Pairing Adapter implements the logic required for obtaining all required data from the External Broker based on the incoming request.
```

Comment:
- ok. Please say what will happen when application is not assigned to any Integration System.  I think we should just return token from connector service, or client-credentials. Please think about that and back to me.

Reply:
Please add API proposal and define contract between Director and PairingAdapter/s.